### PR TITLE
[STG93] Added ACL response headers

### PR DESCRIPTION
**Pull request by tasherif-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/src/Shared/Constants.cs [hunk 1]
```diff
@@ -391,6 +391,8 @@ internal static class DataLake
             public const string GroupHeaderName = "x-ms-group";
 
             public const string PermissionsHeaderName = "x-ms-permissions";
+
+            public const string ACLHeaderName = "x-ms-acl";
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Common/src/Shared/Constants.cs:395]: NIT - even for acronyms it would still be camel case, so "AclHeaderName" is better

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeClientTestFixtureAttribute.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public DataLakeClientTestFixtureAttribute()
                 StorageVersionExtensions.MaxVersion)
         {
             RecordingServiceVersion = StorageVersionExtensions.MaxVersion;
-            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion };
+            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion};
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeClientTestFixtureAttribute.cs:38]: NIT whitespace change

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs [hunk 3]
```diff
@@ -3383,6 +3383,89 @@ public async Task ReadAsync()
...
+                }
+            };
+
+            await fileClient.CreateAsync(options: options);
+
+            // Arrange
+            var data = GetRandomBuffer(Constants.KB);
+            using (var stream = new MemoryStream(data))
+            {
+                await fileClient.AppendAsync(stream, 0);
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs:3409]: Nit: we don't actually need to append or flush data to this file for this test.  We also only need to check the ACLs in the assert.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs [hunk 4]
```diff
@@ -3383,6 +3383,89 @@ public async Task ReadAsync()
...
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs:3456]: Nit: same here, I don't think we actually need to append and flush data in this test.

**Review by seanmcc-msft — COMMENTED** [review]
I have a few nitpicks but looks good to me!

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs (hunk 3 above)
Comment (tasherif-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs:3409]: I know that we don't necessarily need to append and flush, I just felt having some data as a sanity check to read to make sure that ReadAsync method still correctly works under the conditions of appending and flushing data + acls, so I'll just keep it, feel free to reopen this if you think it needs to be removed

**Pull request merged**